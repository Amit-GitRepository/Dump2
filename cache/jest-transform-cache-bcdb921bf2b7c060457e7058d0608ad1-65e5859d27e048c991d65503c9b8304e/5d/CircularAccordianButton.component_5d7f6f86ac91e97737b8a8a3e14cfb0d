Object.defineProperty(exports,"__esModule",{value:true});var cov_1fut6j6ir6=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/ProductCards/BaseProductCollapsibleBar/CircularAccordianButton.component.js',hash='fd7b670023e84346ff657bf79e67f8e33e5ace65',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/ProductCards/BaseProductCollapsibleBar/CircularAccordianButton.component.js',statementMap:{'0':{start:{line:10,column:23},end:{line:10,column:65}},'1':{start:{line:14,column:42},end:{line:14,column:52}},'2':{start:{line:15,column:23},end:{line:15,column:105}},'3':{start:{line:16,column:35},end:{line:16,column:127}},'4':{start:{line:17,column:4},end:{line:22,column:20}},'5':{start:{line:25,column:0},end:{line:29,column:2}},'6':{start:{line:30,column:0},end:{line:34,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:13,column:2},end:{line:13,column:3}},loc:{start:{line:13,column:12},end:{line:23,column:3}},line:13}},branchMap:{'0':{loc:{start:{line:15,column:23},end:{line:15,column:105}},type:'cond-expr',locations:[{start:{line:15,column:38},end:{line:15,column:85}},{start:{line:15,column:88},end:{line:15,column:105}}],line:15},'1':{loc:{start:{line:16,column:35},end:{line:16,column:127}},type:'cond-expr',locations:[{start:{line:16,column:50},end:{line:16,column:100}},{start:{line:16,column:103},end:{line:16,column:127}}],line:16}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0},f:{'0':0},b:{'0':[0,0],'1':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Icon=require('../../../../Shared/Icon/Icon.component');var _Icon2=_interopRequireDefault(_Icon);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _CircularAccordianButton=require('./CircularAccordianButton.style');var _CircularAccordianButton2=_interopRequireDefault(_CircularAccordianButton);
var _Touchable=require('../../../../Shared/Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _reactNative=require('react-native');
var _reactNativeAnimatable=require('react-native-animatable');var Animatable=_interopRequireWildcard(_reactNativeAnimatable);function _interopRequireWildcard(obj){if(obj&&obj.__esModule){return obj;}else{var newObj={};if(obj!=null){for(var key in obj){if(Object.prototype.hasOwnProperty.call(obj,key))newObj[key]=obj[key];}}newObj.default=obj;return newObj;}}function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}

var AnimatableIcon=(cov_1fut6j6ir6.s[0]++,Animatable.createAnimatableComponent(_Icon2.default));var

CircularAccordianButton=function(_Component){_inherits(CircularAccordianButton,_Component);function CircularAccordianButton(){_classCallCheck(this,CircularAccordianButton);return _possibleConstructorReturn(this,(CircularAccordianButton.__proto__||Object.getPrototypeOf(CircularAccordianButton)).apply(this,arguments));}_createClass(CircularAccordianButton,[{key:'render',value:function render()
{cov_1fut6j6ir6.f[0]++;var _ref=(cov_1fut6j6ir6.s[1]++,
this.props),isCollapsed=_ref.isCollapsed,onClick=_ref.onClick,style=_ref.style;
var arrowStyle=(cov_1fut6j6ir6.s[2]++,!isCollapsed?(cov_1fut6j6ir6.b[0][0]++,[_CircularAccordianButton2.default.collapsedArrowStyle,_CircularAccordianButton2.default.arrowStyle]):(cov_1fut6j6ir6.b[0][1]++,_CircularAccordianButton2.default.arrowStyle));
var circularContainerStyle=(cov_1fut6j6ir6.s[3]++,!isCollapsed?(cov_1fut6j6ir6.b[1][0]++,[_CircularAccordianButton2.default.circularContainer,_CircularAccordianButton2.default.closedContainer]):(cov_1fut6j6ir6.b[1][1]++,_CircularAccordianButton2.default.circularContainer));cov_1fut6j6ir6.s[4]++;
return(
_react2.default.createElement(_Touchable2.default,{onPress:onClick,style:[_CircularAccordianButton2.default.container,style]},
_react2.default.createElement(_reactNative.View,{style:circularContainerStyle},
_react2.default.createElement(AnimatableIcon,{name:'chevron-down',style:arrowStyle,transition:'rotate',useNativeDriver:true}))));


}}]);return CircularAccordianButton;}(_react.Component);cov_1fut6j6ir6.s[5]++;

CircularAccordianButton.defaultProps={
isCollapsed:true,
onClick:_noop2.default,
style:{}};cov_1fut6j6ir6.s[6]++;

CircularAccordianButton.propTypes={
isCollapsed:_propTypes2.default.bool,
onClick:_propTypes2.default.func,
style:_propTypes2.default.object};exports.default=


CircularAccordianButton;