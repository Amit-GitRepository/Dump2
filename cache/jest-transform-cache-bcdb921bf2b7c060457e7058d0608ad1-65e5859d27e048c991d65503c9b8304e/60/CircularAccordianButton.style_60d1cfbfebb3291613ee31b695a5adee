Object.defineProperty(exports,"__esModule",{value:true});var cov_1o0wcgffs=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/ProductCards/BaseProductCollapsibleBar/CircularAccordianButton.style.js',hash='42cb478df62915ac9c5b09e74d71e771a7b6b145',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/ProductCards/BaseProductCollapsibleBar/CircularAccordianButton.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../../../themes/constants.styles');exports.default=

{
container:{
position:'absolute',
bottom:0,
transform:[{translateX:-16}]},

closedContainer:{
borderWidth:0},

circularContainer:{
height:20,
width:40,
marginBottom:-1,
borderTopLeftRadius:40,
borderTopRightRadius:40,
borderWidth:1,
borderColor:_constants.colors.FRENCH_GREY,
backgroundColor:_constants.colors.GRAY_NURSE,
alignItems:'center',
justifyContent:'center'},

collapsedArrowStyle:{
transform:[{rotate:'180deg'}]},

arrowStyle:{
color:_constants.colors.PRIMARY_ACTIONABLE,
fontSize:_constants.fonts.FONT_SIZE_SMALL}};