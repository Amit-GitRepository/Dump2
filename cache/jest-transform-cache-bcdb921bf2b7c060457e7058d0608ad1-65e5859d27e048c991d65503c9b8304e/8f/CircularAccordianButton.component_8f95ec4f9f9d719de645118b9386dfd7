Object.defineProperty(exports,"__esModule",{value:true});var cov_1fut6j6ir6=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/ProductCards/BaseProductCollapsibleBar/CircularAccordianButton.component.js',hash='f23926c665b2b4cde751a7211a6dd2037c608199',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/ProductCards/BaseProductCollapsibleBar/CircularAccordianButton.component.js',statementMap:{'0':{start:{line:9,column:23},end:{line:9,column:65}},'1':{start:{line:13,column:42},end:{line:13,column:52}},'2':{start:{line:14,column:23},end:{line:14,column:105}},'3':{start:{line:15,column:35},end:{line:15,column:127}},'4':{start:{line:16,column:4},end:{line:21,column:20}},'5':{start:{line:24,column:0},end:{line:28,column:2}},'6':{start:{line:29,column:0},end:{line:33,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:12,column:2},end:{line:12,column:3}},loc:{start:{line:12,column:12},end:{line:22,column:3}},line:12}},branchMap:{'0':{loc:{start:{line:14,column:23},end:{line:14,column:105}},type:'cond-expr',locations:[{start:{line:14,column:38},end:{line:14,column:85}},{start:{line:14,column:88},end:{line:14,column:105}}],line:14},'1':{loc:{start:{line:15,column:35},end:{line:15,column:127}},type:'cond-expr',locations:[{start:{line:15,column:50},end:{line:15,column:100}},{start:{line:15,column:103},end:{line:15,column:127}}],line:15}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0},f:{'0':0},b:{'0':[0,0],'1':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _CircularAccordianButton=require('./CircularAccordianButton.style');var _CircularAccordianButton2=_interopRequireDefault(_CircularAccordianButton);
var _Shared=require('../../../../Shared');
var _reactNative=require('react-native');
var _reactNativeAnimatable=require('react-native-animatable');var Animatable=_interopRequireWildcard(_reactNativeAnimatable);function _interopRequireWildcard(obj){if(obj&&obj.__esModule){return obj;}else{var newObj={};if(obj!=null){for(var key in obj){if(Object.prototype.hasOwnProperty.call(obj,key))newObj[key]=obj[key];}}newObj.default=obj;return newObj;}}function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}

var AnimatableIcon=(cov_1fut6j6ir6.s[0]++,Animatable.createAnimatableComponent(_Shared.Icon));var

CircularAccordianButton=function(_Component){_inherits(CircularAccordianButton,_Component);function CircularAccordianButton(){_classCallCheck(this,CircularAccordianButton);return _possibleConstructorReturn(this,(CircularAccordianButton.__proto__||Object.getPrototypeOf(CircularAccordianButton)).apply(this,arguments));}_createClass(CircularAccordianButton,[{key:'render',value:function render()
{cov_1fut6j6ir6.f[0]++;var _ref=(cov_1fut6j6ir6.s[1]++,
this.props),isCollapsed=_ref.isCollapsed,onClick=_ref.onClick,style=_ref.style;
var arrowStyle=(cov_1fut6j6ir6.s[2]++,!isCollapsed?(cov_1fut6j6ir6.b[0][0]++,[_CircularAccordianButton2.default.collapsedArrowStyle,_CircularAccordianButton2.default.arrowStyle]):(cov_1fut6j6ir6.b[0][1]++,_CircularAccordianButton2.default.arrowStyle));
var circularContainerStyle=(cov_1fut6j6ir6.s[3]++,!isCollapsed?(cov_1fut6j6ir6.b[1][0]++,[_CircularAccordianButton2.default.circularContainer,_CircularAccordianButton2.default.closedContainer]):(cov_1fut6j6ir6.b[1][1]++,_CircularAccordianButton2.default.circularContainer));cov_1fut6j6ir6.s[4]++;
return(
_react2.default.createElement(_Shared.Touchable,{onPress:onClick,style:[_CircularAccordianButton2.default.container,style]},
_react2.default.createElement(_reactNative.View,{style:circularContainerStyle},
_react2.default.createElement(AnimatableIcon,{name:'chevron-down',style:arrowStyle,transition:'rotate',useNativeDriver:true}))));


}}]);return CircularAccordianButton;}(_react.Component);cov_1fut6j6ir6.s[5]++;

CircularAccordianButton.defaultProps={
isCollapsed:true,
onClick:_noop2.default,
style:{}};cov_1fut6j6ir6.s[6]++;

CircularAccordianButton.propTypes={
isCollapsed:_propTypes2.default.bool,
onClick:_propTypes2.default.func,
style:_propTypes2.default.object};exports.default=


CircularAccordianButton;