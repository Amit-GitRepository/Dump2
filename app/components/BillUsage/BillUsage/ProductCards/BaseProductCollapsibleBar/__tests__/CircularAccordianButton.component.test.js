import CircularAccordianButton from '../CircularAccordianButton.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('CircularAccordianButton component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<CircularAccordianButton />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
