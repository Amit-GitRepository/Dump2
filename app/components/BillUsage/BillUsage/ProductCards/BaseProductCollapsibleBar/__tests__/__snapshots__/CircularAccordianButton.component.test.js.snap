// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularAccordianButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "bottom": 0,
      "height": 21,
      "opacity": 1,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateX": -16,
        },
      ],
      "width": 40,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Image
        getSize={[Function]}
        isBackground={false}
        onLayout={[Function]}
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/assets/semiCircle.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
